zerver/data_import/slack_message_conversion.py

```diff
@@ -157,13 +157,19 @@ def convert_markdown_syntax(text: str, regex: str, zulip_keyword: str) -> str:
 def convert_link_format(text: str) -> tuple[str, bool]:
     """
     1. Converts '<https://foo.com>' to 'https://foo.com'
-    2. Converts '<https://foo.com|foo>' to 'https://foo.com|foo'
+    2. Converts '<https://foo.com|foo>' to '[foo](https://foo.com)'
     """
     has_link = False
     for match in re.finditer(LINK_REGEX, text, re.VERBOSE):
-        converted_text = match.group(0).replace(">", "").replace("<", "")
+        slack_url = match.group(0)
+        url_parts = slack_url[1:-1].split("|", maxsplit=1)
+        converted_url = url_parts[0]
+        # Check if there's a pipe with text after it
+        if len(url_parts) == 2:
+            converted_url = f"[{url_parts[1]}]({url_parts[0]})"
```

Comment:
I think an if/else would be easier to read here as we only expect two potential cases ...

```python
        url_parts = slack_url[1:-1].split("|", maxsplit=1)
        if len(url_parts) == 1:
            converted_url = url_parts[0]
        else:
            converted_url = f"[{url_parts[1]}]({url_parts[0]})"
```